Packs/ANYRUN/Integrations/ANYRUN/ANYRUN_test.py

```diff
@@ -128,3 +134,30 @@ def test_taskid_from_url(self):
         url = 'https://www.madeup.com/madeup/tasks/'  # disable-secrets-detection
         url += 'this-is-the-task-id/blah/&someotherstuff'
         assert taskid_from_url(url) == 'this-is-the-task-id'
+
+
+def http_mock(method, url, verify=USE_SSL, headers=None):
+    if not headers:
+        raise ValueError("Permission denied")
+    else:
+        the_response = requests.Response()
+        the_response.status_code = 200
+        the_response._content = b'{ "image" : "image" }'
+        return the_response
+
+
+def test_get_image(mocker):
+    """
+    Given:
+        A mock response from the API
+    When:
+        Calling images_from_report
+    Then:
+        Validate that the http request gets called with authorization (headers)
+    """
+    response = {'data': {'analysis': {'content': {'screenshots': [{'permanentUrl': 'www.test.com'}]}}}}
+    mocker.patch('requests.request', side_effect=http_mock)
```

Comment:
you can use `requests_mock`. see `docs/tutorial-integration/YodaSpeak/Integrations/YodaSpeak/YodaSpeak_test.py:35`

Reply:
At first I used requests_mock, but I did not found how to check if the request had "headers" sent with it, so I changed back to mocker.patch

Reply:
https://requests-mock.readthedocs.io/en/latest/matching.html#request-headers